specs/deneb/polynomial-commitments-sampling.md

```diff
@@ -0,0 +1,530 @@
+
+- [Introduction](#introduction)
+- [Custom types](#custom-types)
+- [Constants](#constants)
+- [Preset](#preset)
+  - [Samples](#samples)
+  - [Crypto](#crypto)
+- [Helper functions](#helper-functions)
+  - [Linear combinations](#linear-combinations)
+    - [`g2_lincomb`](#g2_lincomb)
+  - [FFTs](#ffts)
+    - [`_simple_ft_field`](#_simple_ft_field)
+    - [`_fft_field`](#_fft_field)
+    - [`fft_field`](#fft_field)
+  - [Polynomials in coefficient form](#polynomials-in-coefficient-form)
+    - [`polynomial_eval_to_coeff`](#polynomial_eval_to_coeff)
+    - [`polynomial_coeff_to_eval`](#polynomial_coeff_to_eval)
+    - [`add_polynomialcoeff`](#add_polynomialcoeff)
+    - [`neg_polynomialcoeff`](#neg_polynomialcoeff)
+    - [`multiply_polynomialcoeff`](#multiply_polynomialcoeff)
+    - [`divide_polynomialcoeff`](#divide_polynomialcoeff)
+    - [`shift_polynomialcoeff`](#shift_polynomialcoeff)
+    - [`interpolate_polynomialcoeff`](#interpolate_polynomialcoeff)
+    - [`zero_polynomialcoeff`](#zero_polynomialcoeff)
+    - [`evaluate_polynomialcoeff`](#evaluate_polynomialcoeff)
+  - [KZG multiproofs](#kzg-multiproofs)
+    - [`compute_kzg_proof_multi_impl`](#compute_kzg_proof_multi_impl)
+    - [`verify_kzg_proof_multi_impl`](#verify_kzg_proof_multi_impl)
+  - [Sample cosets](#sample-cosets)
+    - [`coset_for_sample`](#coset_for_sample)
+- [Samples](#samples-1)
+  - [Sample computation](#sample-computation)
+    - [`compute_samples_and_proofs`](#compute_samples_and_proofs)
+    - [`compute_samples`](#compute_samples)
+  - [Sample verification](#sample-verification)
+    - [`verify_sample_proof`](#verify_sample_proof)
+    - [`verify_sample_proof_batch`](#verify_sample_proof_batch)
+- [Reconstruction](#reconstruction)
+  - [`recover_samples_impl`](#recover_samples_impl)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document extends [polynomial-commitments.md](polynomial-commitments.md) with the functions required for data availability sampling (DAS). It is not part of the core Deneb spec but an extension that can be optionally implemented to allow nodes to reduce their load using DAS.
+
+For any KZG library extended to support DAS, functions flagged as "Public method" MUST be provided by the underlying KZG library as public functions. All other functions are private functions used internally by the KZG library.
+
+Public functions MUST accept raw bytes as input and perform the required cryptographic normalization before invoking any internal functions.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `PolynomialCoeff` | `Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB]` | A polynomial in coefficient form |
+
+## Constants
+
+| Name | Value | Notes |
+| - | - | - |
+
+## Preset
+
+### Samples
+
+| Name | Value | Description |
+| - | - | - |
+| `FIELD_ELEMENTS_PER_SAMPLE` | `uint64(64)` | Number of field elements in a sample |
```

Comment:
We are changing PeerDAS to 1D construction: https://github.com/hwwhww/consensus-specs/pull/1/

Reply:
I still think consistent naming is good, and the cryptography spec does not really know about samples either. To be completely neutral it would also be possible to call it "coset".